gobblin-runtime/src/main/java/gobblin/runtime/MysqlDatasetStateStore.java

```diff
@@ -64,12 +94,57 @@ public MysqlDatasetStateStore(DataSource dataSource, String stateStoreTableName,
    * @throws IOException if there's something wrong reading the {@link JobState.DatasetState}s
    */
   public Map<String, JobState.DatasetState> getLatestDatasetStatesByUrns(String jobName) throws IOException {
-    List<JobState.DatasetState> previousDatasetStates =
-        getAll(jobName, "%" + CURRENT_DATASET_STATE_FILE_SUFFIX + DATASET_STATE_STORE_TABLE_SUFFIX, true);
+    final String storeName = jobName;
```

Comment:
Mainly with the duplication. If there is no interface at the StateStore level that makes sense then having the query logic at the dataset level is fine, but the existing implementation calls getAll() with a pattern. Can't a similar API be provided that will get all items with the pattern and having the most recent modification time?